tests/factorization/test_implicit.py

```diff
@@ -23,6 +23,39 @@ def _min_max_scale(arr):
     return (arr - arr_min) / (arr_max - arr_min)
 
 
+@pytest.mark.parametrize('weights, loss, expected', [
```

Comment:
Two comments:
1. Can we instantiate the weights arrays inside the function? You could use one of `(0, 1)` as a scaling factor in the pytest args, and do the multiplication (with `np.ones`) inside the test function. I think that would make it cleaner.
2. These tests will run forever :) Maybe we could run them on a smaller subset of the data?